arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/MonoConsumer.java

```diff
@@ -14,17 +14,23 @@ public MonoConsumer(DynamicClassExtractor extractor) {
         this.extractor = extractor;
     }
 
+    /**
+     * support for Mono type recording
+     * @param responseMono
+     * @return
+     */
     public Mono<?> accept(Mono<?> responseMono) {
         return responseMono
-            .doOnSuccess(o -> {
-                try (TraceTransmitter tm = traceTransmitter.transmit()) {
-                    extractor.recordResponse(o);
-                }
+            .doOnSuccess(result -> {
+                traceTransmitter.transmit();
+                extractor.recordResponse(result);
+                traceTransmitter.close();
             })
-            .doOnError(o -> {
-                try (TraceTransmitter tm = traceTransmitter.transmit()) {
-                    extractor.recordResponse(o);
-                }
```

Comment:
see [The try-with-resources Statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)